creation/interfaces-and-networks.md

```diff
@@ -578,13 +578,22 @@ not automatic; it should be configured via cloud init, as shown below:
         pod: {}
       volumes:
       - cloudInitNoCloud:
+          networkData: |
+            version: 2
+            ethernets:
+              eth0:
+                addresses: [ fd10:0:2::2/120 ]
+                dhcp4: true
```

Comment:
Do we need to mention this for IPv6 clusters?

Reply:
If you want to keep having an IPv4 address, yes. 

Do we want that ?

Reply:
We only support single-stack IPv4 or v6. Having v4 that is not functional may be confusing on cluster that has only IPv6 addresses.

Reply:
That is already there - the VMs will have an ipv4 address configured via dhcp - our current implementation provides the interface an ipv4 address. That is the source of confusion, not this, imho.

This is just explicitly setting the IP address - so users don't become suprised when they log into the VM and find a useless, non functional IPv4 address set there.

Reply:
It gets IPv4 since there is IPv4 DHCP client implicitly running in OS. But here you recommend to explicitly enable it.

> This is just explicitly setting the IP address - so users don't become suprised when they log into the VM and find a useless, non functional IPv4 address set there.

That's what I'd like to avoid by not having this option. But I don't care too much about it, since it is not a regression.

Reply:
> It gets IPv4 since there is IPv4 DHCP client implicitly running in OS. But here you recommend to explicitly enable it.

I do recommend to explicitly enable it because that's the current behavior. It would be disabled by default without this config line, which leads to the absence of the currently useless-but-existing IPv4 address. 

Furthermore, I honestly believe that explicit is better than implicit. As a user that wants IPv6 on the VMs I would be seriously puzzled from the useless ipv4 address that sneaked into my VM. This way, our dear hypothetical user is not puzzled - he/she literally asked for it. Lastly, when the useless IPv4 address is not desired, the hypothetical user can just omit that line.

I would prefer to not clarify that `dhcpv4: true` is optional, simply because I would have to explain that there is a useless IP address being set on the vm because "we're preparing for dual stack". 

> 
> > This is just explicitly setting the IP address - so users don't become suprised when they log into the VM and find a useless, non functional IPv4 address set there.
> 
> That's what I'd like to avoid by not having this option. But I don't care too much about it, since it is not a regression.

Does this mean you're OK with it as is then?

Reply:
> I honestly believe that explicit is better than implicit.
+1

I think either enable or disable dhcp4 is ok as long as you explain the unexpected (enable dhcp I guess).

In the kubevirt vmi example there is a good reasoning to have dhcpv4: the example works on both ipv4 and ipv6 setups.
In the tests, we are ready for dual stack.

Here it will be a bit harder to reason for all the above and the existence of a previous unexplained (and implicit) behavior is not sufficient. I leave it to you to decide what is better, just make a good comment in the commit if yiu decide to keep the dhcpv4 as true (I do not think there is a need to explain when it is false).